include/unified_teleop/controller.hpp

```diff
@@ -0,0 +1,124 @@
+        
+        public:
+            //
+            // CONSTRUCTORS
+            //
+            /// @brief Default constructor for Controller.
+            Controller();
+
+            /// @brief Constructor that initializes the Controller with device configuration.
+            /// @param device_config_file_path YAML Node containing the device configuration file path.
+            Controller(YAML::Node device_config_file_path);
+
+            //
+            // FUNCTIONS
+            //
+
+            /// @brief Reads and updates the current joystick state.
+            /// @param msg The latest joystick message received.
+            void update_joy_state(const sensor_msgs::msg::Joy::SharedPtr &msg);
+
+            /// @brief Reads the device configuration file and updates the button map accordingly.
+            /// @param device_config_file_path YAML Node containing the device configuration file path.
+            void update_button_map(YAML::Node device_config_file_path);
+
+            /// @brief Generates a MovementInput object based on the input name and current button map.
+            /// @param input_assignment The name of the input to generate a MovementInput for.
+            /// @return An optional MovementInput object.
+            std::optional<MovementInput> generate_MovementInput(const std::string input_assignment);
+
+            /// @brief Reads the state of a given input from a given joy state.
+            /// @param input The MovementInput object to read the state for.
+            /// @param joy_state The provided joy state.
+            /// @return The state of the input as a double.
+            double read_MovementInput(const std::optional<MovementInput> input, const sensor_msgs::msg::Joy joy_state);
+
+            /// @brief Clears the vector of MovementInput objects.
+            void clear_MovementInput_vec();
+
+            /// @brief Checks if the current joy state is different from the previous one.
+            /// @return True if different, otherwise false.
+            bool is_joy_state_different();
+
+            /// @brief Checks if a given input is enabled.
+            /// @param input The MovementInput object to check.
+            /// @return True if the input is enabled or if the variable always_enabled is true, otherwise false.
+            bool is_enabled(const std::optional<rosnu::MovementInput> input);
+
+            //
+            // GETTERS
+            //
+
+            /// @brief Gets the input type based on the input name.
+            /// @param input_name The name of the input.
+            /// @return The InputType of the given input name.
+            InputType get_input_type(std::string input_name);
+
+            /// @brief Gets the current joystick state.
+            /// @return The current joy state.
+            sensor_msgs::msg::Joy get_current_joy_state();
+
+            //
+            // SETTERS
+            //
+
+            /// @brief Sets the always_enabled flag.
```

Comment:
Why is this settable and not just set in the constructor

Reply:
Because variables such as input/output config and always_enabled parameter are only assigned in the node initializer

Reply:
So I need it to be a setter otherwise I'd need to construct the controller object before I have a chance to read parameters in the init